### PR TITLE
Update Gen3 Renesas programming procedure

### PR DESCRIPTION
**Pull request by abinayaddhandapani**
Update the Renesas programming procedure as per the RENESAS GENERATION 3.5 PROGRAMMING GUIDE.

Update the Infineon TDA programming procedure as per the AN_2308_163442-V1.2(TDA38xxxRev5 Programming Guide)

Add PMBUS address and Slave address in the Congo and Morocco config file.

Test fields: Verified in Congo system

### CODE REVIEW

File: inc/vr_update_renesas_gen3.hpp [hunk 1]
```diff
@@ -10,12 +10,12 @@
 #define FW_WRITE             (0xE6)
 #define DMA_WRITE             0xC7
 #define DMA_READ             (0xC5)
-#define GEN3_CRC_ADDR        0x0094
+#define GEN3_CRC_ADDR        (0x00F8)
```
Comment (ojayanth) [inline inc/vr_update_renesas_gen3.hpp:13]: Why these Addresses are getting changed? is it specific to version change ?

Comment (abinayaddhandapani) [reply inc/vr_update_renesas_gen3.hpp:13]: These are the new registers to be accessed for the new parts in Venice

File: src/vr_update_infineon_tda.cpp [hunk 2]
```diff
@@ -279,9 +250,11 @@ bool vr_update_infineon_tda::UpdateFirmware()
                             index_range = std::stoi(word, nullptr, BASE_16);
                             if(index_range < INDEX_40)
                                 break;
-                            else if(index_range > INDEX_70 && index_range < INDEX_200)
+                            else if(index_range >= 0x80 && index_range < INDEX_200)
```
Comment (supven01) [inline src/vr_update_infineon_tda.cpp]: magic index range 0x80

File: src/vr_update_infineon_tda.cpp [hunk 3]
```diff
@@ -279,9 +250,11 @@ bool vr_update_infineon_tda::UpdateFirmware()
...
                                 break;
-                            else if(index_range > INDEX_2FF)
+                            else if(index_range >= 0x300 && index_range < 0x380)
```
Comment (supven01) [inline src/vr_update_infineon_tda.cpp]: magic numbers

File: src/vr_update_infineon_tda.cpp [hunk 4]
```diff
@@ -279,9 +250,11 @@ bool vr_update_infineon_tda::UpdateFirmware()
...
+                                break;
+                            else if(index_range >= 0x390)
```
Comment (supven01) [inline src/vr_update_infineon_tda.cpp]: magic number

**Review by supven01 — CHANGES_REQUESTED** [review]

File: src/vr_update_infineon_tda.cpp (hunk 2 above)
Comment (abinayaddhandapani) [reply src/vr_update_infineon_tda.cpp]: Ack

File: src/vr_update_infineon_tda.cpp (hunk 3 above)
Comment (abinayaddhandapani) [reply src/vr_update_infineon_tda.cpp]: Ack

File: src/vr_update_infineon_tda.cpp (hunk 4 above)
Comment (abinayaddhandapani) [reply src/vr_update_infineon_tda.cpp]: Ack

**Review by supven01 — APPROVED** [review]

**Pull request merged**